### ISSUE TITLE
Cannot read secret due to data store in `["data"]["data"]`, but not `["data"]`

### ISSUE THREAD

**Issue by czchen** [tomtom-international/vault-anyconfig#15]
* vault-anyconfig version: 0.2.0
* Python version: 3.7
* Operating System: Debian sid

### Description

* When we try to read secret in vault, we got the following error. It looks like vault stores data in `["data"]["data"]`, but `vault-anyconfig` reads data in `["data"]` only. We are not sure it is caused by using wrong way to store our secret or not.

```python
    def __vault_keys_retrieve(self, config):
        """
        Connects to the Vault to retrieve keys specified in the config dictionary
    
        Args:
            - config: configuration dict
        Returns:
            Updated vault configuration pieces
        """
        vault_config_parts = {}
    
        if self.pass_through_flag:
            warn(
                "VaultAnyconfig is set to Passthrough mode, but secrets are configured in configuration. These secrets will not be loaded.",                                                                      
                UserWarning,
            )
            return vault_config_parts
    
        for secret, path in config.get("vault_secrets", {}).items():
            config_key_path = secret.split(".")
            secret_path = path
    
            # Optionally map the key in the configuration to a different key in the Vault
            secret_path_split = secret_path.split(".")
            if len(secret_path_split) > 1:
                secret_path = "".join(secret_path_split[:-1])
                secret_key = secret_path_split[-1]
            else:
                secret_key = config_key_path[-1]
    
>           read_vault_secret = self.read(secret_path)["data"][secret_key]
E           KeyError: 'SQLALCHEMY_DATABASE_URI'

<censored>/lib/python3.7/site-packages/vault_anyconfig/vault_anyconfig.py:237: KeyError                                                                          
>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>> entering PDB >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
> <censored>/lib/python3.7/site-packages/vault_anyconfig/vault_anyconfig.py(237)__vault_keys_retrieve()                                                      
-> read_vault_secret = self.read(secret_path)["data"][secret_key]
(Pdb) p self.read(secret_path)["data"]
Replacing double-slashes ("//") in path with single slash ("/") to avoid Vault redirect response.
{'data': {'<censored>': '<censored>'}, 'metadata': {'created_time': '2019-05-13T02:50:34.02335485Z', 'deletion_time': '', 'destroyed': False, 'version': 3}}

* We use `vault kv put path key=value` to store our secret

```

### What I Did

```
```python
    def __vault_keys_retrieve(self, config):
        """
        Connects to the Vault to retrieve keys specified in the config dictionary
    
        Args:
            - config: configuration dict
        Returns:
            Updated vault configuration pieces
        """
        vault_config_parts = {}
    
        if self.pass_through_flag:
            warn(
                "VaultAnyconfig is set to Passthrough mode, but secrets are configured in configuration. These secrets will not be loaded.",                                                                      
                UserWarning,
            )
            return vault_config_parts
    
        for secret, path in config.get("vault_secrets", {}).items():
            config_key_path = secret.split(".")
            secret_path = path
    
            # Optionally map the key in the configuration to a different key in the Vault
            secret_path_split = secret_path.split(".")
            if len(secret_path_split) > 1:
                secret_path = "".join(secret_path_split[:-1])
                secret_key = secret_path_split[-1]
            else:
                secret_key = config_key_path[-1]
    
>           read_vault_secret = self.read(secret_path)["data"][secret_key]
E           KeyError: 'SQLALCHEMY_DATABASE_URI'
```

**Comment by eugene-davis** [comment]
The return from Vault is structured `["data"][key]`, where `data` is all of the secrets stored at a given path.

Based on the error you have provided, it is accessing `SQLALCHEMY_DATABASE_URI` but that key is not set on the given path in Vault.

If I had to guess, you setup the secret with something like `vault kv put secret/SQLALCHEMY_DATABASE_URI SQLALCHEMY_DATABASE_URI=<value>` and in `vault_secrets` you are provide the path as `secret/SQLALCHEMY_DATABASE_URI` rather than `secret/SQLALCHEMY_DATABASE_URI.SQLALCHEMY_DATABASE_URI` but I don't have enough information to be sure.

**Comment by czchen** [comment]
@eugene-davis ,

We set secret with command like `vault kv put /secret/config SQLALCHEMY_DATABASE_URI=<value>`. Maybe there are some problems in vault? Or how we use vault?

**Comment by eugene-davis** [comment]
Can you configure that `vault kv get secret/config` returns your secret with `SQLALCHEMY_DATABASE_URI` listed as a key?

Also can you provide a (sanitized as needed) copy of the `vault_secrets` section you are using?

**Comment by czchen** [comment]
* The following is our vault output

```
$ vault kv get /secret/config
====== Metadata ======
Key              Value
---              -----
created_time     2019-05-13T02:50:34.02335485Z
deletion_time    n/a
destroyed        false
version          3

============= Data =============
Key                        Value
---                        -----
SQLALCHEMY_DATABASE_URI    <censored>
```

* The following is `vault_secrets` section in our config file.

```
vault_secrets:
  'app.config.SQLALCHEMY_DATABASE_URI': /secret/config.SQLALCHEMY_DATABASE_URI
```

**Comment by eugene-davis** [comment]
Please try:
```
vault_secrets:
  app.config.SQLALCHEMY_DATABASE_URI: "secret/config.SQLALCHEMY_DATABASE_URI"
```

So far I haven't been able to replicate the nested data array issue you have seen, can you tell me what version the Vault server is?

**Comment by mangokingTW** [comment]
Vault has two versions of KV secret engine.
v1: https://www.vaultproject.io/api/secret/kv/kv-v1.html#sample-request
v2: https://www.vaultproject.io/api/secret/kv/kv-v2.html#sample-request-2 
It looks like vault-anyconfig can only support v1.
Is it possible to support both two versions?

**Comment by eugene-davis** [comment]
The fix is in 0.2.1
